Graphics/Slicer/Math/PGA.hs

```diff
@@ -135,12 +135,29 @@ intersectsWith (Right pl1) (Right pl2) = Right $ plinesIntersectIn   pl1 pl2
 intersectsWith (Left l1)   (Right pl1) =         lineIntersectsPLine l1  pl1
 intersectsWith (Right pl1) (Left l1)   =         lineIntersectsPLine l1  pl1
 
+-- | Check if/where two line segments intersect.
+lineIntersection :: LineSeg -> LineSeg -> Either Intersection PIntersection
+lineIntersection l1@(LineSeg p1 s1) l2@(LineSeg p2 s2)
+  | meet2PLine2 (eToPLine2 l1) (eToPLine2 l2) == PPoint2 (GVec [])         = Right PColinear
+  | onSegment l1 intersection && onSegment l2 intersection && intersection == p1 = Left $ HitStartPoint l1 intersection
```

Comment:
I'd suggest pulling `onSegment l1 intersection && onsegment l2 intersection` into a where-binding, which would eliminate most of the duplicating here.

Reply:
good idea.